nifi-nar-bundles/nifi-influxdb-bundle/nifi-influxdb-processors/src/main/java/org/apache/nifi/processors/influxdb/AbstractInfluxDBProcessor.java

```diff
@@ -0,0 +1,143 @@
+            .build();
+
+    public static final PropertyDescriptor USERNAME = new PropertyDescriptor.Builder()
+            .name("influxdb-username")
+            .displayName("Username")
+            .required(false)
+            .description("Username for accessing InfluxDB")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor PASSWORD = new PropertyDescriptor.Builder()
+            .name("influxdb-password")
+            .displayName("Password")
+            .required(false)
+            .description("Password for user")
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .sensitive(true)
+            .build();
+
+    protected static final PropertyDescriptor MAX_RECORDS_SIZE = new PropertyDescriptor.Builder()
+            .name("influxdb-max-records-size")
+            .displayName("Max size of records")
+            .description("Maximum size of records allowed to be posted in one batch")
+            .defaultValue("1 MB")
+            .required(true)
+            .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
+            .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+            .description("Sucessful FlowFiles are routed to this relationship").build();
+
+    static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+            .description("Failed FlowFiles are routed to this relationship").build();
+
+    public static final String INFLUX_DB_ERROR_MESSAGE = "influxdb.error.message";
+
+    protected InfluxDB influxDB;
+    protected long maxRecordsSize;
+
+    /**
+     * Helper method to help testability
+     * @return InfluxDB instance
+     */
+    protected InfluxDB getInfluxDB() {
+        return influxDB;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        String username = context.getProperty(USERNAME).getValue();
+        String password = context.getProperty(PASSWORD).getValue();
+        String influxDbUrl = context.getProperty(INFLUX_DB_URL).getValue();
+
+        maxRecordsSize = context.getProperty(MAX_RECORDS_SIZE).asDataSize(DataUnit.B).longValue();
+
+        try {
+            influxDB = makeConnection(username, password, influxDbUrl);
+        } catch(Exception e) {
+            getLogger().error("Error while getting connection {}", new Object[] { e.getLocalizedMessage() },e);
+            throw new RuntimeException("Error while getting connection" + e.getLocalizedMessage(),e);
+        }
+        getLogger().info("InfluxDB connection created for host {}",
+                new Object[] {influxDbUrl});
+    }
+
+    protected InfluxDB makeConnection(String username, String password, String influxDbUrl) {
+        if ( StringUtils.isBlank(username) || StringUtils.isBlank(password) ) {
+            return InfluxDBFactory.connect(influxDbUrl);
```

Comment:
In order to set a connection timeout on the underlying OkHttp client (see my other comments), you'll need to create that client yourself and pass it into the [InfluxDbFactory.connect](https://github.com/influxdata/influxdb-java/blob/influxdb-java-2.8/src/main/java/org/influxdb/InfluxDBFactory.java#L59) method